src/object/cli_obj.c

```diff
@@ -3106,7 +3108,7 @@ obj_shard_comp_cb(struct shard_auxi_args *shard_auxi,
 		if (ret != -DER_REC2BIG && !obj_retry_error(ret) &&
 		    !obj_is_modification_opc(obj_auxi->opc) &&
 		    !obj_auxi->is_ec_obj && !obj_auxi->spec_shard &&
-		    !obj_auxi->to_leader) {
+		    !obj_auxi->spec_group && !obj_auxi->to_leader) {
```

Comment:
spec_group actually should be different as spec_shard in this case. Since spec_group still need retry, i.e. it needs to do the following check if there are replicas available.  
But this is probably not this patch issue.